src/proper_gen_next.erl

```diff
@@ -60,7 +60,9 @@
 -define(GEN_NEXT_DEPTH,  proper_gen_next_cache_depth).
 
 -type temperature() :: float().
--type nf() :: fun((pos_integer(), temperature()) -> term()).
+-type depth() :: pos_integer().
+-type nf_temp() :: temperature() | {depth(), temperature()} | null.
+-type nf() :: fun((any(), nf_temp()) -> proper_types:type()).
```

Comment:
The `nf()` fun changed type for its first argument from `pos_integer()` to `any()`.
1. Is this correct?  Was it wrong before?
2. If it indeed can accept any term in its first arg, the type should read `term()` there as in the type below.

Reply:
1. Yes this was wrong, as the `nf()` is the neighborhood function provided by the user. It's signature is `fun (Base, Temp) -> ...`, where `Base` depends on the initial `RawType`, so this is not a `pos_integer()` for sure.
2. Yes, will update to `term()`.